### ISSUE TITLE
Loss of base and large models

### ISSUE THREAD

**Issue by stefan-it** [google-research/electra#3]
Hi,

I'm currently working on a new non-English ELECTRA model. Training on GPU seems to work and is running fine 🤗

Next steps would be to try model training on a TPU, so I would just like to ask if you can post the final loss of both base and large models (or even share the loss training curve) so that we have a kind of reference point when training own models 🤔

Thanks many in advance,

Stefan

**Comment by clarkkev** [comment]
Hi Stefan,

Great to see non-English models being trained already! Here are some pre-training cures (x-axis is pre-train steps) for the models. Note that these metrics were computed with `"do_eval": true` which means there is no dropout, so training losses will be slightly higher. 

Small model on OpenWebText, batch_size=128, mask_percent=0.15
![image](https://user-images.githubusercontent.com/1091306/76335698-256fb500-62b2-11ea-9fee-e39aca5cae24.png)

Base model on WikiBooks, batch_size=256, mask_percent=0.15
![image](https://user-images.githubusercontent.com/1091306/76336223-ea21b600-62b2-11ea-88f3-046b02686839.png)

Large model on XLNet data, batch_size=2048, mask_percent=0.25
![image](https://user-images.githubusercontent.com/1091306/76335591-02dd9c00-62b2-11ea-82b8-de2a600a9ea9.png)

The losses will vary depending on your setting because the disciminator loss depends on the quality of the generator, the generator loss can depend on the training data., etc.

**Comment by stefan-it** [comment]
Hi @clarkkev,

thanks for these interesting insights! Great to have these training curves 🤗

**Comment by yaolu** [comment]
@clarkkev Do you have any idea about why there's a peak in disc_loss at the very beginning of the training curve?

**Comment by jarednielsen** [comment]
@yaolu From our own experiments, the generator starts out creating random predictions, then plateaus for a bit using the median prediction. So every [MASK] gets replaced with the token `the` or the `.` token. This makes the discriminator's job very easy for a bit. Eventually the generator moves past this plateau and starts generating more diverse predictions, which causes the discriminator loss to peak for a bit.

Here's a sample validation prediction at step 1000. The red-highlighted tokens are the generator's replacements, and the yellow-highlighted tokens are the tokens which the discriminator predicts as corrupted. Note that the discriminator simply predicts that every `the` and `.` token is corrupt, and the generator in this case replaces every `[MASK]` with a `.`.

<img width="1358" alt="Screen Shot 2020-07-08 at 2 34 10 PM" src="https://user-images.githubusercontent.com/4564897/86972671-27df0000-c128-11ea-94ae-918e3e698452.png">

**Comment by blueseasky** [comment]
Hi,
    Great to have these training curves! I'm currently working on a new non-English ELECTRA large model. I encountered some problems during pre-training. The loss first drops after a period of time and then rises. I would just like to ask if you can post the learning rate of both base and large models (or even share the learning rate curve) so that we have a kind of reference point about learning rate when training own models . The following is my loss and acc curve. Looking forward to your suggestions.

![image](https://user-images.githubusercontent.com/29243072/103730414-c7b91000-501d-11eb-9286-7fb6a3367183.png)

Thanks many in advance!

**Comment by Jimojimojimo** [comment]
> Hi,
> Great to have these training curves! I'm currently working on a new non-English ELECTRA large model. I encountered some problems during pre-training. The loss first drops after a period of time and then rises. I would just like to ask if you can post the learning rate of both base and large models (or even share the learning rate curve) so that we have a kind of reference point about learning rate when training own models . The following is my loss and acc curve. Looking forward to your suggestions.
> 
> ![image](https://user-images.githubusercontent.com/29243072/103730414-c7b91000-501d-11eb-9286-7fb6a3367183.png)
> 
> Thanks many in advance!

How did you draw the graph? 
When I set the do_eval as true and evaluated the model, it was marked as only one point when I checked with Tensorboard. 
How can I check the graph every step?

**Comment by zshy1205** [comment]
> as only one poin
I also can not draw this graph, are you solve it?
